lib/msf/core/web_services/views/auth/login.erb

```diff
@@ -7,20 +7,56 @@
 </head>
 <body>
 
+<script
+  src="https://code.jquery.com/jquery-3.3.1.min.js"
+  integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8="
+  crossorigin="anonymous"></script>
+<script>
+    function ConvertFormToJSON(form){
+        var array = jQuery(form).serializeArray();
+        var json = {};
+
+        jQuery.each(array, function() {
+            json[this.name] = this.value || '';
+        });
+
+        return JSON.stringify(json);
+    }
+
+    $(document).ready(function(){
+        $("#submit").on('click', function(){
+            $.ajax({
+                url: '<%= AuthServlet.api_generate_token_path %>',
+                type : "POST",
+                dataType : 'json',
+                data : ConvertFormToJSON($("#login_form")),
+                contentType: "application/json; charset=utf-8",
+                processData: false,
+                success : function(result) {
+                    $(location).attr('href', '<%= AuthServlet.api_account_path %>')
+                },
+                error: function(xhr, resp, text) {
+                    console.log(xhr, resp, text);
+                }
+            })
+        });
+    });
+
+</script>
+
 <div style="padding:20px;">
-  <form action="<%= AuthServlet.api_login_path %>" method="post">
+  <form id="login_form" action="" method="post">
     <div class="credential-container">
       <h2>Log In - Metasploit API</h2>
       <label for="username"><b>Username</b></label>
       <input type="text" placeholder="Enter Username" name="username" required>
 
       <label for="password"><b>Password</b></label>
       <input type="password" placeholder="Enter Password" name="password" required>
-
-      <button type="submit">Log In</button>
+      <input id="submit" type="button" name="submit" value="submit">
```

Comment:
Minor: the button text was changed from "Log In"

Reply:
In addition, `lib/msf/core/web_services/public/msf-ws.css` contains styling for the button so that it filled the bottom of the log on credential container `div`. Update that selector so that it applies to this new input button.